sdk/servicebus/Azure.Messaging.ServiceBus/src/Sender/ServiceBusSender.cs

```diff
@@ -489,8 +494,14 @@ private void InstrumentMessages(IEnumerable<ServiceBusMessage> messages)
         {
             Argument.AssertNotDisposed(IsClosed, nameof(ServiceBusSender));
             cancellationToken.ThrowIfCancellationRequested<TaskCanceledException>();
-            var sequenceNumberList = sequenceNumbers.ToArray();
-            Logger.CancelScheduledMessagesStart(Identifier, sequenceNumberList);
+            long[] sequenceNumbersArray = sequenceNumbers.ToArray();
```

Comment:
`ToArray` will also always allocate.   You may want to consider a type sniff if you're not concerned with making a copy.